pkg/git/provider.go

```diff
@@ -17,50 +17,49 @@ package git
 import (
 	"context"
 	"fmt"
+	"net/http"
 	"strings"
 )
 
+//go:generate mockgen -destination=./mocks/roundTripper.go -package=mocks net/http RoundTripper
+
 type (
-	TokenType    string
 	ProviderType string
 
 	// Provider represents a git provider
 	Provider interface {
-		Type() ProviderType
 		BaseURL() string
-		VerifyToken(ctx context.Context, tokenType TokenType, token string) error
 		SupportsMarketplace() bool
+		Type() ProviderType
+		VerifyRuntimeToken(ctx context.Context, token string) error
+		VerifyUserToken(ctx context.Context, token string) error
 	}
 )
 
-const (
-	RuntimeToken  TokenType = "runtime token"
-	PersonalToken TokenType = "personal token"
-)
-
-var providers = map[ProviderType]func(string) (Provider, error){
+var providers = map[ProviderType]func(string, *http.Client) (Provider, error){
 	BITBUCKET_SERVER: NewBitbucketServerProvider,
 	GITHUB:           NewGithubProvider,
 	GITHUB_ENT:       NewGithubProvider, // for backward compatability
 	GITLAB:           NewGitlabProvider,
 }
 
 func GetProvider(providerType ProviderType, baseURL string) (Provider, error) {
+	client := &http.Client{}
```

Comment:
this is where we can add anything we want to the client, to enable retries, or any other such settings